src/libcollections/str.rs

```diff
@@ -10,9 +10,28 @@
 
 //! Unicode string slices.
 //!
+//! The `&str` type is one of the two main string types, the other being `String`.
+//! Unlike its `String` counterpart, its contents are borrowed and therefore
+//! cannot be moved someplace else.
```

Comment:
This is pretty solid, but I would probably end up just ending the sentence after "borrowed", as you can move `&str` values, just not what they point _to_, which is very tricky and not really relevant here, IMHO.